src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1950,5 +1950,24 @@ public void GetPartsOfCastExpression(SyntaxNode node, out SyntaxNode type, out S
 
             return null;
         }
+
+        public bool IsPartOfPropertyDeclarationHeader(SyntaxNode node)
```

Comment:
i would be ok with this just being an internal impl detail of the refactoring selection service.

Reply:
(actually, keep this here, but make a sibling of IsInTypeHeader (and try to unify if we can on hte signatures).

Reply:
I wanted to be consistent with IsOnTypeHeader already being there. And TBH it make sense to me in SyntaxFacts.

Reply:
agreed.  i'm fine with it being in ISyntaxFacts.  Just keep it together with IsInTypeHeader, and ideally name it similarly and make the signatures similar.